src/find/matchers/time.rs

```diff
@@ -0,0 +1,325 @@
+        // given_modification_time (in which case we want to return true) then
+        // duration_since will return an error.
+        Ok(self.given_modification_time.duration_since(this_time).is_err())
+    }
+}
+
+impl Matcher for NewerMatcher {
+    fn matches(&self, file_info: &DirEntry, _: &mut MatcherIO) -> bool {
+        match self.matches_impl(file_info) {
+            Err(e) => {
+                writeln!(&mut stderr(),
+                         "Error getting modification time for {}: {}",
+                         file_info.path().to_string_lossy(),
+                         e)
+                    .unwrap();
+                false
+            }
+            Ok(t) => t,
+        }
+    }
+
+    fn has_side_effects(&self) -> bool {
+        false
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum FileTimeType {
+    Accessed,
+    Created,
+    Modified,
+}
+
+impl FileTimeType {
+    fn get_file_time(self, metadata: Metadata) -> std::io::Result<SystemTime> {
+        match self {
+            FileTimeType::Accessed => metadata.accessed(),
+            FileTimeType::Created => metadata.created(),
+            FileTimeType::Modified => metadata.modified(),
+        }
+    }
+}
+
+/// This matcher checks whether a file's accessed|creation|modification time is
+/// {less than | exactly | more than} N days old.
+pub struct FileTimeMatcher {
+    days: ComparableValue,
+    file_time_type: FileTimeType,
+}
+
+impl Matcher for FileTimeMatcher {
+    fn matches(&self, file_info: &DirEntry, matcher_io: &mut MatcherIO) -> bool {
+        match self.matches_impl(file_info, matcher_io.now()) {
+            Err(e) => {
+                writeln!(&mut stderr(),
+                         "Error getting {:?} time for {}: {}",
+                         self.file_time_type,
+                         file_info.path().to_string_lossy(),
+                         e)
+                    .unwrap();
+                false
+            }
+            Ok(t) => t,
+        }
+    }
+
+    fn has_side_effects(&self) -> bool {
+        false
+    }
+}
+
+
+
+impl FileTimeMatcher {
+    /// Impementation of matches that returns a result, allowing use to use try!
+    /// to deal with the errors.
+    fn matches_impl(&self, file_info: &DirEntry, now: SystemTime) -> Result<bool, Box<Error>> {
+        let this_time = try!(self.file_time_type.get_file_time(try!(file_info.metadata())));
+        let mut is_negative = false;
+        // durations can't be negative. So duration_since returns a duration
+        // wrapped in an error if now < this_time.
+        let age = match now.duration_since(this_time) {
+            Ok(duration) => duration,
+            Err(e) => {
+                is_negative = true;
+                e.duration()
+            }
+        };
+        let age_in_seconds: i64 = age.as_secs() as i64 * if is_negative { -1 } else { 1 };
+        // rust division truncates towards zero (see
```

Comment:
This logic isn't exactly truncation, it's "age in days" - 1 if the age is an exact multiple. Maybe that's ok, but comment doesn't quite match code.